contrib/platform/test/com/sun/jna/platform/mac/SystemBTest.java

```diff
@@ -64,33 +64,39 @@ public void testSysctl() {
         final String mibName = "hw.logicalcpu";
         final int nCpu = Runtime.getRuntime().availableProcessors();
 
-        IntByReference size = new IntByReference(SystemB.INT_SIZE);
-        Pointer p = new Memory(size.getValue());
-        int ret = SystemB.INSTANCE.sysctlbyname(mibName, p, size, null, 0);
-        assertEquals(ret, 0);
+        // This sysctl returns a 32-bit integer cpu count
+        size_t size = new size_t(Integer.BYTES);
+        Pointer p = new Memory(size.longValue());
+        size_t.ByReference plen = new size_t.ByReference(size);
+        assertEquals(0, SystemB.INSTANCE.sysctlbyname(mibName, p, plen, null, size_t.ZERO));
         // These values should be equal unless affinity is set, limiting nCpu
         assertTrue(p.getInt(0) >= nCpu);
 
-        size = new IntByReference();
-        ret = SystemB.INSTANCE.sysctlnametomib(mibName, null, size);
-        assertEquals(ret, 0);
-        // Size should be 2
-        assertEquals(size.getValue(), 2);
-
-        Pointer mibp = new Memory(size.getValue() * SystemB.INT_SIZE);
-        ret = SystemB.INSTANCE.sysctlnametomib(mibName, mibp, size);
-        assertEquals(ret, 0);
-        // Size should be 2
-        assertEquals(size.getValue(), 2);
-
-        int[] mib = mibp.getIntArray(0, size.getValue());
+        // Get the same value by converting the string to the MIB array
+        // Get the size of the MIB array
+        size = new size_t(Integer.BYTES);
+        size_t.ByReference sizep = new size_t.ByReference(size);
+        assertEquals(0, SystemB.INSTANCE.sysctlnametomib(mibName, null, sizep));
```

Comment:
I assume `sizep` is intented to take the required size of the memory, then the size is not needed, but only `sizep` and `sizep` should be initialized to zero.